analyzers/tests/SonarAnalyzer.UnitTest/Rules/Utilities/TokenTypeAnalyzerTest.Classifier.cs

```diff
@@ -222,4 +222,677 @@ public class Test {
+    [DataRow("{ [u:InnerException].[u:InnerException].[u:Data]: {} }", false)] // in SubpatternSyntax.ExpressionColon context. Must be properties
+    public void IdentifierToken_SimpleMemberAccess_ExpressionColon(string pattern, bool allowSemanticModel) =>
+        // found in SubpatternSyntax.ExpressionColon
+        ClassifierTestHarness.AssertTokenTypes($$"""
+            using System;
+            using System.Collections.Generic;
+            public class C
+            {
+                public void M(Exception ex)
+                {
+                    _ = ex is {{pattern}};
+                }
+            }
+            """/*, allowSemanticModel */);
+
+    [DataTestMethod]
+    [DataRow("[t:Int32]", true)]
+    [DataRow("[u:i]", true)]
+    [DataRow("[t:Int32].[u:MaxValue]", true)]
+    [DataRow("[u:System].[t:Int32]", true)]
+    [DataRow("not [t:Int32]", true)]
+    [DataRow("not [u:i]", true)]
+    [DataRow("not [t:Int32].[u:MaxValue]", true)]
+    [DataRow("not [u:System].[t:Int32]", true)]
+    public void IdentifierToken_SimpleMemberAccess_Is(string pattern, bool allowSemanticModel) =>
+        ClassifierTestHarness.AssertTokenTypes($$"""
+            using System;
+            using System.Collections.Generic;
+            public class C
+            {
+                public void M(object o)
+                {
+                    const int i = 42;
+                    _ = o is {{pattern}};
+                }
+            }
+            """, allowSemanticModel);
+
+    [DataTestMethod]
+    [DataRow("[t:Int32]", true)]
+    [DataRow("[u:i]", true)]
+    [DataRow("[t:Int32].[u:MaxValue]", true)]
+    [DataRow("[u:System].[t:Int32].[u:MaxValue]", true)]
+    [DataRow("not [t:Int32]", true)]
+    [DataRow("not [u:i]", true)]
+    [DataRow("not [t:Int32].[u:MaxValue]", true)]
+    [DataRow("not [u:System].[t:Int32]", true)]
+    public void IdentifierToken_SimpleMemberAccess_SwitchArm(string pattern, bool allowSemanticModel) =>
+        ClassifierTestHarness.AssertTokenTypes($$"""
+            using System;
+            using System.Collections.Generic;
+            public class C
+            {
+                public void M(object o)
+                {
+                    const int i = 42;
+                    switch(o)
+                    {
+                        case {{pattern}}: break;
+                    }
+                }
+            }
+            """, allowSemanticModel);
+
+    [DataTestMethod]
+    [DataRow("[t:Int32]", true)]
+    [DataRow("[u:i]", true)]
+    [DataRow("[t:Int32].[u:MaxValue]", true)]
+    [DataRow("[u:System].[t:Int32].[u:MaxValue]", true)]
+    [DataRow("not [t:Int32]", true)]
+    [DataRow("not [u:i]", true)]
+    [DataRow("not [t:Int32].[u:MaxValue]", true)]
+    [DataRow("not [u:System].[t:Int32]", true)]
+    public void IdentifierToken_SimpleMemberAccess_SwitchExpression(string pattern, bool allowSemanticModel) =>
```

Comment:
should we add some cases with "where" too?

Reply:
Did you mean "when"? If so, "when" is a "normal" expression context. We may need to add some more test cases for such expression context (unary, binary, invocation, ternary, etc.). I would say that his is a pre-condition for the sprint to be closed, but I don't think we need each and every expression context at the moment.
What is special about the IdentifierToken_SimpleMemberAccess_SwitchExpression, IdentifierToken_SimpleMemberAccess_SwitchArm, and IdentifierToken_SimpleMemberAccess_Is test cases is that here, types and constant expressions are allowed at the same time. Therefore we always need to query the semantic model here.